core/unix/os.c

```diff
@@ -8502,35 +8557,35 @@ get_dynamo_library_bounds(void)
     check_start = dynamo_dll_start;
 #endif /* STATIC_LIBRARY */
 
-    static char dynamorio_libname_buf[MAXIMUM_PATH];
-    res = memquery_library_bounds(NULL, &check_start, &check_end, dynamorio_library_path,
-                                  BUFFER_SIZE_ELEMENTS(dynamorio_library_path),
-                                  dynamorio_libname_buf,
-                                  BUFFER_SIZE_ELEMENTS(dynamorio_libname_buf));
+    if (do_memquery) {
+        static char dynamorio_libname_buf[MAXIMUM_PATH];
+        res = memquery_library_bounds(
+            NULL, &check_start, &check_end, dynamorio_library_path,
+            BUFFER_SIZE_ELEMENTS(dynamorio_library_path), dynamorio_libname_buf,
+            BUFFER_SIZE_ELEMENTS(dynamorio_libname_buf));
 #ifndef STATIC_LIBRARY
-    dynamorio_libname = IF_UNIT_TEST_ELSE(UNIT_TEST_EXE_NAME, dynamorio_libname_buf);
+        dynamorio_libname = IF_UNIT_TEST_ELSE(UNIT_TEST_EXE_NAME, dynamorio_libname_buf);
 #endif /* STATIC_LIBRARY */
 
-    LOG(GLOBAL, LOG_VMAREAS, 1, PRODUCT_NAME " library path: %s\n",
-        dynamorio_library_path);
-    snprintf(dynamorio_library_filepath, BUFFER_SIZE_ELEMENTS(dynamorio_library_filepath),
-             "%s%s", dynamorio_library_path, dynamorio_libname);
-    NULL_TERMINATE_BUFFER(dynamorio_library_filepath);
-    LOG(GLOBAL, LOG_VMAREAS, 1, PRODUCT_NAME " library file path: %s\n",
-        dynamorio_library_filepath);
-    NULL_TERMINATE_BUFFER(dynamorio_library_filepath);
+        LOG(GLOBAL, LOG_VMAREAS, 1, PRODUCT_NAME " library path: %s\n",
+            dynamorio_library_path);
+        snprintf(dynamorio_library_filepath,
+                 BUFFER_SIZE_ELEMENTS(dynamorio_library_filepath), "%s%s",
+                 dynamorio_library_path, dynamorio_libname);
+        NULL_TERMINATE_BUFFER(dynamorio_library_filepath);
+        LOG(GLOBAL, LOG_VMAREAS, 1, PRODUCT_NAME " library file path: %s\n",
+            dynamorio_library_filepath);
+        NULL_TERMINATE_BUFFER(dynamorio_library_filepath);
 #if !defined(STATIC_LIBRARY) && defined(LINUX)
-    ASSERT(check_start == dynamo_dll_start && check_end == dynamo_dll_end);
+        ASSERT(check_start == dynamo_dll_start && check_end == dynamo_dll_end);
 #elif defined(MACOS)
-    ASSERT(check_start == dynamo_dll_start);
-    dynamo_dll_end = check_end;
+        ASSERT(check_start == dynamo_dll_start);
+        dynamo_dll_end = check_end;
 #else
-    dynamo_dll_start = check_start;
-    dynamo_dll_end = check_end;
+        dynamo_dll_start = check_start;
+        dynamo_dll_end = check_end;
 #endif
-    LOG(GLOBAL, LOG_VMAREAS, 1, "DR library bounds: " PFX " to " PFX "\n",
-        dynamo_dll_start, dynamo_dll_end);
-    ASSERT(res > 0);
```

Comment:
Good catch on both accounts! I meant to move the LOG to the end (so that it happens for all cases) and that didn't happen, and I definitely didn't meant to delete this assert. (although I have no idea why it was all the way down here, so I put it right after the memquery_library_bounds.)